lib/youtube-dl.js

```diff
@@ -234,7 +234,13 @@ ytdl.getInfo = function(url, args, options, callback) {
 
 
 var formatsRegex =
-  /^(\d+)\s+([a-z0-9]+)\s+(\d+x\d+|\d+p|audio only|\?x\d+|unknown)/;
+  /^([\w\- ]+?)\s+([a-z0-9]{2,4})\s+(\d+x\d+|\d+p|audio only|\?x\d+|unknown)(.*)$/;
+var optionalSpecsRegex =
+  /\s+([^,]*?)\s+(\d+k)(.*)$/;
+var fpsRegex = 
+  /, (\d+)fps,/;
+var sizeRegex =
+  /, (\d+\.?\d{0,2}[G|M|K]iB)/;
```

Comment:
the optional modifier should be applied to the whole right side of the decimal right? something like `(?:\.\d{1,2})?`

and you don't need to put the pipes inside of the character sets

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#special-character-set